src/actions/account.js

```diff
@@ -305,16 +324,28 @@ export const { addAccessKey, createAccountWithSeedPhrase, addAccessKeySeedPhrase
     ],
     CREATE_ACCOUNT_WITH_SEED_PHRASE: [
         async (accountId, recoveryKeyPair, fundingContract, fundingKey) => {
+            const previousAccountId = wallet.accountId
+
             await wallet.saveAccount(accountId, recoveryKeyPair);
             await wallet.createNewAccount(accountId, fundingContract, fundingKey, recoveryKeyPair.publicKey)
+           
             const publicKey = recoveryKeyPair.publicKey.toString()
             const contractName = null;
             const fullAccess = true;
             const newKeyPair = KeyPair.fromRandom('ed25519')
             const newPublicKey = newKeyPair.publicKey
-            await wallet.addAccessKey(accountId, contractName, newPublicKey, fullAccess)
-            await wallet.saveAccount(accountId, newKeyPair)
+            
             await wallet.postSignedJson('/account/seedPhraseAdded', { accountId, publicKey })
+            try {
+                await wallet.addAccessKey(accountId, contractName, newPublicKey, fullAccess)
+                await wallet.saveAccount(accountId, newKeyPair)
+            } catch (error) {
+                wallet.clearAccountState()
```

Comment:
Let's not ever call `.clearAccountState()`. Definitely not on _any_ error. Means we are removing local keys if we get timeout error or smth like that.

Reply:
@marcinbodnar I think it's best to remove this method altogether. We can instead have some UI to indicate when account doesn't exist on blockchain / or there are no local keys.

Reply:
OK, but in this case, we want to clear the account that was added few lines higher - so the user is able to recover this account once again. We can probably skip `delete this.accounts[this.accountId]` from `clearAccountState` but we need to run `this.accountId = ''` anyway. Is that ok?

Reply:
> we want to clear the account that was added few lines higher - so the user is able to recover this account once again.

Not sure I understand the problem. Why should recovery fail if we don't clear account state? 

> but we need to run `this.accountId = ''` anyway

Why is this needed unless we know for sure that account creation failed (e.g. account with such name was created before)?

Reply:
Recovery will work but it might be a bit weird for the user. He would be asked to recover the account but at the same time, he will be logged in on this account.

Reply:
Something being a little weird is definitely better alternative than user losing keys :)

Reply:
of course :) but there is an additional issue - for example:
 - if a user is creating an account
 - the key was not added
 - user is redirected to `/recover-seed-phrase`
 - user skips it - he's logged in so it's possible
 - user add email recovery
 - user delete seed phrase recovery
 - both recovery options are deleted.

Not sure why this is happening, but I guess it's because email recovery is created with the main access key (not recovery seed phrase access key, as it does not exist). And this might cause serious problems because the user might not notice it and ends up with no recovery at all.

My idea was to force a recovering account, but if that's forbidden, I think we need to figure out something else to avoid this scenario.

Reply:
The problem here is that when user is creating account on mainnet – they are claiming linkdrop vs just creating account for free. So if anything goes wrong – there is potential of real money being lost.
If we "force recovery" as you suggest, we can easily end up in situation where all keys to account are lost. 

We need to avoid this dire situation at all cost. This includes recovering from half-broken states, even if it is harder than wiping everything.

Reply:
Many users already lost their mainnet accounts because not writing down seed phrase when they got error – so this is not hypothetical situation.

Reply:
yes, I understand it completely.
The only thing is this issue:

> * if a user is creating an account
> * the key was not added
> * user is redirected to `/recover-seed-phrase`
> * user skips it - he's logged in so it's possible
> * user add email recovery
> * user delete seed phrase recovery
> * both recovery options are deleted.

Maybe it's unlikely, but I think we should still handle it. My proposition:
 - if this situation appears, instead of showing that seed phrase is added (on `/profile`) we will show info that some error appears and a button to fix it, that will redirect the user to `/recover-seed-phrase`